dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -826,6 +827,144 @@ ShowStatusPropertyDialog(
     }
 }
 
+VOID ShowNetworkIconContextMenu(
+    _In_ HWND hwndOwner,
+    _In_ LANSTATUSUI_CONTEXT *pContext)
+{
+    if (!pContext || !pContext->pNet)
+        return;
+
+    HMENU hMenu = CreatePopupMenu();
+    if (!hMenu) return;
+
+    POINT pt;
+    GetCursorPos(&pt);
+
+    SetForegroundWindow(hwndOwner);
+
+    // The context menu items, set to their default values.
+    struct
+    {
+        UINT uID;
+        UINT uFlags;
+        UINT_PTR uIDNewItem; // wCmdID
+    } MenuItems[] =
+    {
+        {IDS_NET_ACTIVATE, MF_STRING | MF_GRAYED, IDM_NETICON_ENABLE},
+        {IDS_NET_STATUS, MF_STRING | MF_GRAYED, IDM_NETICON_STATUS},
+        {IDS_NET_REPAIR, MF_STRING | MF_GRAYED, IDM_NETICON_REPAIR},
+        {-1, 0, -1}, // Separator
+        {IDS_NET_OPEN_CONNECTIONS, MF_STRING, IDM_NETICON_OPEN_CONNECTIONS},
+        {IDS_NET_PROPERTIES, MF_STRING | MFS_DEFAULT, IDM_NETICON_PROPERTIES},
+    };
+
+    NETCON_PROPERTIES *pProps = NULL;
+    HRESULT hr = pContext->pNet->GetProperties(&pProps);
+    if (SUCCEEDED(hr) && pProps)
+    {
+        if (pProps->Status == NCS_HARDWARE_DISABLED ||
+            pProps->Status == NCS_MEDIA_DISCONNECTED ||
+            pProps->Status == NCS_DISCONNECTED)
+        {
+            MenuItems[0].uID = IDS_NET_ACTIVATE;
+            MenuItems[0].uFlags = MF_STRING | MFS_DEFAULT;
+            MenuItems[0].uIDNewItem = IDM_NETICON_ENABLE;
+            MenuItems[5].uFlags = MF_STRING;
+        }
+        else
+        {
+            MenuItems[0].uID = IDS_NET_DEACTIVATE;
+            MenuItems[0].uFlags = MF_STRING;
+            MenuItems[0].uIDNewItem = IDM_NETICON_DISABLE;
+        }
+
+        if (pProps->Status == NCS_CONNECTED)
+        {
+            MenuItems[1].uFlags = MF_STRING;
+            MenuItems[2].uFlags = MF_STRING;
+        }
+        else if (pProps->Status == NCS_CONNECTING)
+        {
+            MenuItems[1].uFlags = MF_STRING;
+            MenuItems[2].uFlags = MF_STRING | MF_GRAYED;
+        }
+        else
+        {
+            MenuItems[1].uFlags = MF_STRING | MF_GRAYED;
+            MenuItems[2].uFlags = MF_STRING | MF_GRAYED;
+        }
+
+        NcFreeNetconProperties(pProps);
+        pProps = NULL;
+    }
+    else
+    {
+        MenuItems[0].uFlags = MF_STRING | MF_GRAYED;
+        MenuItems[1].uFlags = MF_STRING | MF_GRAYED;
+        MenuItems[2].uFlags = MF_STRING | MF_GRAYED;
+        MenuItems[5].uFlags = MF_STRING | MF_GRAYED;
+    }
+
+    // Set the "Properties" item as default, if the Network "Enable/Disable" item isn't.
+    if (!(MenuItems[0].uFlags & MFS_DEFAULT))
+        MenuItems[5].uFlags |= MFS_DEFAULT;
+
+    WCHAR szMenuItem[128];
+
+    for (USHORT i = 0; i < _countof(MenuItems); ++i)
+    {
+        if (MenuItems[i].uID != -1)
```

Comment:
```suggestion
        if (MenuItems[i].uID != UINT_MAX)
```
fixing compiler yapping there as well.